pkg/plugins/resource/resource.go

```diff
@@ -0,0 +1,171 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	clusterapiv1 "open-cluster-management.io/api/cluster/v1"
+	clusterapiv1alpha1 "open-cluster-management.io/api/cluster/v1alpha1"
+	"open-cluster-management.io/placement/pkg/plugins"
+)
+
+const (
+	placementLabel = "cluster.open-cluster-management.io/placement"
+	description    = `
+	ResourceRatio[ResourceType] and ResourceAllocatable[ResourceType] prioritizer makes the scheduling 
+	decisions based on the resource allocatable to capacity ratio or allocatable of managed clusters. 
+	The [ResourceType] could be CPU or Memory.
+	The clusters that has the most allocatable to capacity ratio or allocatable are given the highest score, 
+	while the least is given the lowest score.
+	`
+)
+
+var _ plugins.Prioritizer = &Resource{}
+
+type Resource struct {
+	handle   plugins.Handle
+	resource string
+	mode     string
+}
+
+type ResourceBuilder struct {
+	resource *Resource
+}
+
+func NewResourceBuilder(handle plugins.Handle) *ResourceBuilder {
```

Comment:
how about NewResourcePrioritizerBuilder(handle).WithPrioritizerName("resourceCPUAllocatable").Build()

and you can maintain a map here
```go
resourceMap := map[string]clusterapiv1.ResourceName {
    "CPU":  clusterapiv1.ResourceCPU
    ...
}

func (r *ResourcePrioritizerBuilder) Build() {
    resource, mode := parsePrioritizerName(r.prioritizerName)
    return &resourcePrioritizer{
        resource: resourceMap[resource], // or this map find is put in parsePrioritizerName
        mode: mode,
        name: r.prioritizerName,
    }
}
```

Reply:
Updated.